### ISSUE TITLE
How to change the Nix channel?

### ISSUE THREAD

**Issue by chris-martin** [travis-ci/travis-ci#8979]
The [Nix documentation](https://docs.travis-ci.com/user/languages/nix) says

> The default channel for `nixpkgs` will be `nixpkgs-unstable`.

which implies that there is some way in mind to change that to a different channel. Can we add that to the documentation?

Side note - I'm not sure unstable is a good choice of default? It causes a fair bit of build breakage; for example, [Stack is currently not building on unstable](https://github.com/NixOS/nixpkgs/issues/33097).

cc @domenkozar @garbas @matthewbauer

**Comment by chris-martin** [comment]
Best recommendation I have so far is:

```yaml
before_install:
  - nix-channel --add https://nixos.org/channels/nixos-17.09 nixpkgs
  - nix-channel --update
```

**Comment by matthewbauer** [comment]
Yes that's the best way to do it now. We could make it customizable I suppose in Travis but this is just the default channel in Nix:

https://github.com/NixOS/nix/blob/27788f406078489b2fc04749ebb86e267a19a6a5/scripts/install-darwin-multi-user.sh#L578

**Comment by turboMaCk** [comment]
Are you sure this solution works? I'm kind of lost. A project that was building fine 3 days ago stopped working and I simply can't make it work. I've spent quite some time trying to figure it out but no luck https://travis-ci.org/turboMaCk/agile-poker/branches

**Comment by turboMaCk** [comment]
got it... it has to be set as a channel for `nixpkgs` not `nixos` of course :X
